src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -167,22 +224,197 @@ private float[] getMinScores(final List<CompoundTopDocs> queryTopDocs, final int
         return minScores;
     }
 
-    private float normalizeSingleScore(final float score, final float minScore, final float maxScore) {
+    private float normalizeSingleScore(final float score, final float minScore, final float maxScore, final LowerBound lowerBound) {
         // edge case when there is only one score and min and max scores are same
         if (Floats.compare(maxScore, minScore) == 0 && Floats.compare(maxScore, score) == 0) {
             return SINGLE_RESULT_SCORE;
         }
-        float normalizedScore = (score - minScore) / (maxScore - minScore);
-        return normalizedScore == 0.0f ? MIN_SCORE : normalizedScore;
+        if (lowerBound.isEnabled() == false) {
+            return LowerBound.Mode.IGNORE.normalize(score, minScore, maxScore, lowerBound.getMinScore());
+        }
+        return lowerBound.getMode().normalize(score, minScore, maxScore, lowerBound.getMinScore());
+    }
+
+    /**
+     * Get lower bounds from input parameters
+     * @param params user provided input parameters for this technique
+     * @return optional list of lower bounds. Can be empty in case lower bounds are not provided
+     */
+    private Optional<List<Pair<LowerBound.Mode, Float>>> getLowerBounds(final Map<String, Object> params) {
+        // validate that the input parameters are in correct format
+        if (Objects.isNull(params) || params.containsKey(PARAM_NAME_LOWER_BOUNDS) == false) {
+            return Optional.empty();
+        }
+
+        Object lowerBoundsObj = params.get(PARAM_NAME_LOWER_BOUNDS);
+        if (lowerBoundsObj instanceof List<?> == false) {
+            throw new IllegalArgumentException("lower_bounds must be a List");
+        }
+        List<?> lowerBoundsParams = (List<?>) lowerBoundsObj;
+        // number of lower bounds must match the number of sub-queries in a hybrid query
+        if (lowerBoundsParams.size() > MAX_NUMBER_OF_SUB_QUERIES) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "lower_bounds size %d should be less than or equal to %d",
```

Comment:
For my understanding
Don't we have to support lower bound for every sub query? For the ones we don't we can have mode as `ignore`. That should mean the lower bound size should be equal to sub query size.

Reply:
yes, lower bound is for every sub-query. Here we're cutting extremely large collection with lower bounds even before we start parsing them. For hybrid query we can have up to 5 sub-queries, and here we're checking if lower bounds list does not have more then 5 elements.